core/services/relay/evm/read/event.go

```diff
@@ -548,6 +548,12 @@ func (b *EventBinding) encodeComparator(comparator *primitives.Comparator) (quer
 }
 
 func (b *EventBinding) encodeValComparatorDataWord(dwTypeID string, value any) (hash common.Hash, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("%w: cannot encode %s data word comparator", commontypes.ErrInvalidType, dwTypeID)
```

Comment:
Why are we expecting a panic?

Reply:
I added this, geth abi package in some scenarios panics instead of throwing an error

Reply:
How about mentioning "recovered from panic" somewhere in the returned error? And `r` is critical to include as well.

Reply:
Updated the error message accordingly.